vhost-user-backend/src/event_loop.rs

```diff
@@ -150,49 +218,30 @@ where
     /// associated with the backend.
     pub(crate) fn run(&self) -> VringEpollResult<()> {
         const EPOLL_EVENTS_LEN: usize = 100;
-        let mut events = vec![EpollEvent::new(EventSet::empty(), 0); EPOLL_EVENTS_LEN];
-
-        'epoll: loop {
-            let num_events = match self.epoll.wait(-1, &mut events[..]) {
-                Ok(res) => res,
-                Err(e) => {
-                    if e.kind() == io::ErrorKind::Interrupted {
-                        // It's well defined from the epoll_wait() syscall
-                        // documentation that the epoll loop can be interrupted
-                        // before any of the requested events occurred or the
-                        // timeout expired. In both those cases, epoll_wait()
-                        // returns an error of type EINTR, but this should not
-                        // be considered as a regular error. Instead it is more
-                        // appropriate to retry, by calling into epoll_wait().
-                        continue;
-                    }
-                    return Err(VringEpollError::EpollWait(e));
-                }
-            };
-
-            for event in events.iter().take(num_events) {
-                let evset = match EventSet::from_bits(event.events) {
-                    Some(evset) => evset,
-                    None => {
-                        let evbits = event.events;
-                        println!("epoll: ignoring unknown event set: 0x{evbits:x}");
-                        continue;
-                    }
-                };
 
-                let ev_type = event.data() as u16;
+        let mut events = Events::with_capacity(EPOLL_EVENTS_LEN);
+        'poll: loop {
+            self.poller
+                .lock()
+                .unwrap()
+                .poll(&mut events, None)
+                .map_err(VringPollError::PollerWait)?;
 
-                // handle_event() returns true if an event is received from the exit event fd.
-                if self.handle_event(ev_type, evset)? {
-                    break 'epoll;
+            for event in events.iter() {
+                let token = event.token();
+
+                if let Some(evt_set) = event_to_event_set(event) {
+                    if self.handle_event(token.0 as u16, evt_set)? {
```

Comment:
Since `token.0` is a `usize`, are we sure this cannot be truncated?

Reply:
In handle_event, the data is u16. But in register_event, the data is u64. Maybe we should all use u16 or u64?

Reply:
> > nit: maybe we can move these changes in a separate commit (same PR, before adding `mio`)
> 
> It might not be moved to a separate commit. Because `Epoll::new(events: EventSet, data: u64)`, so we can't change `u64->u16` in `register_event` before using `mio`

why not? 
I mean, for sure we need other changes, but why it depends on `mio`?

Reply:
> > > nit: maybe we can move these changes in a separate commit (same PR, before adding `mio`)
> > 
> > 
> > It might not be moved to a separate commit. Because `Epoll::new(events: EventSet, data: u64)`, so we can't change `u64->u16` in `register_event` before using `mio`
> 
> why not? I mean, for sure we need other changes, but why it depends on `mio`?

Not depend on `mio`, just need some extra changes, like: `EpollEvent::new(ev_type, data as u64)`, `self.handle_event(ev_type, evset.try_into().unwarp())`